project/src/main/world/creature/creature-curve.gd

```diff
@@ -75,8 +87,40 @@ func set_curve_defs(new_curve_defs: Array) -> void:
 	_refresh_curve()
 
 
+"""
+Disconnects creature visuals listeners specific to arm shadows.
+
+Can be overridden to disconnect additional listeners.
+"""
+func disconnect_creature_visuals_listeners() -> void:
+	creature_visuals.disconnect("visual_fatness_changed", self, "_on_CreatureVisuals_visual_fatness_changed")
+	creature_visuals.disconnect("orientation_changed", self, "_on_CreatureVisuals_orientation_changed")
+
+
+"""
+Connects creature visuals listeners specific to arm shadows.
+
+Can be overridden to connect additional listeners.
+"""
+func connect_creature_visuals_listeners() -> void:
+	creature_visuals.connect("visual_fatness_changed", self, "_on_CreatureVisuals_visual_fatness_changed")
+	creature_visuals.connect("orientation_changed", self, "_on_CreatureVisuals_orientation_changed")
+
+
+func _refresh_creature_visuals_path() -> void:
+	if not (is_inside_tree() and creature_visuals_path):
+		return
+	
+	if creature_visuals:
+		disconnect_creature_visuals_listeners()
+	
+	creature_visuals = get_node(creature_visuals_path)
+	
+	connect_creature_visuals_listeners()
```

Comment:
This code doesn't need to be double-spaced; I think this is organized this way because it used to have more lines of code which were grouped, but now it's just a few.

Reply:
Removed spacing.